$NetBSD: patch-CMakeLists.txt,v 1.4 2024/11/19 14:58:05 nia Exp $

- Find exact Lua version with pkg-config.
- Use system dependencies also when compiling the libretro core.

--- CMakeLists.txt.orig	2024-10-21 14:13:00.000000000 +0000
+++ CMakeLists.txt
@@ -462,7 +462,8 @@ if(NOT LIBRETRO)
 		target_link_libraries(${PROJECT_NAME} PRIVATE ZLIB::ZLIB)
 	endif()
 
-	find_package(Lua 5.2)
+	find_package(PkgConfig)
+	pkg_check_modules(LUA lua-5.2)
 	if(NOT APPLE AND LUA_FOUND)
 		target_compile_definitions(${PROJECT_NAME} PRIVATE USE_LUA)
 		target_include_directories(${PROJECT_NAME} PRIVATE ${LUA_INCLUDE_DIR} core/deps/luabridge/Source)
@@ -515,22 +516,22 @@ if(PKG_CONFIG_FOUND AND NOT ANDROID AND 
 		target_link_libraries(${PROJECT_NAME} PRIVATE PkgConfig::LIBPULSE)
 	endif()
 
-	if(USE_HOST_LIBZIP)
-		pkg_check_modules(LIBZIP IMPORTED_TARGET libzip)
-		if(LIBZIP_FOUND)
-			target_link_libraries(${PROJECT_NAME} PRIVATE PkgConfig::LIBZIP)
-		endif()
-	endif()
-
 	if(ENABLE_OPROFILE)
 		target_compile_definitions(${PROJECT_NAME} PRIVATE DYNA_OPROF)
 		target_link_libraries(${PROJECT_NAME} PRIVATE opagent)
 	endif()
+endif()
+
+find_package(MiniUPnPc)
+if(MINIUPNP_FOUND)
+	target_include_directories(${PROJECT_NAME} PRIVATE ${MINIUPNP_INCLUDE_DIRS})
+	target_link_libraries(${PROJECT_NAME} PRIVATE ${MINIUPNP_LIBRARIES})
+endif()
 
-	find_package(MiniUPnPc)
-	if(MINIUPNP_FOUND)
-		target_include_directories(${PROJECT_NAME} PRIVATE ${MINIUPNP_INCLUDE_DIRS})
-		target_link_libraries(${PROJECT_NAME} PRIVATE ${MINIUPNP_LIBRARIES})
+if(USE_HOST_LIBZIP)
+	pkg_check_modules(LIBZIP IMPORTED_TARGET libzip)
+	if(LIBZIP_FOUND)
+		target_link_libraries(${PROJECT_NAME} PRIVATE PkgConfig::LIBZIP)
 	endif()
 endif()
 
